protocols/v2/noise-sv2/src/signature_message.rs

```diff
@@ -24,6 +49,13 @@ impl From<[u8; 74]> for SignatureNoiseMessage {
 }
 
 impl SignatureNoiseMessage {
+    // Verifies the [`SignatureNoiseMessage`] against the provided public key and an optional
+    // authority public key. The verification checks that the message is currently valid
+    // (i.e., within the `valid_from` and `not_valid_after` time window) and that the signature
+    // is correctly signed by the authority.
+    //
+    // If an authority public key is not provided, the function assumes that the signature
```

Comment:
this optional `authority_pk` feels like a hack for testing or something..maybe it should be addressed in the refactor

Reply:
The reason we have an optional `authority_pk` is that we can encounter two possible scenarios: authenticated and unauthenticated connections. When roles are on the same network, it is acceptable to forgo the static key check. However, if they are not on the same network, the static key check becomes mandatory.

Reply:
Ah ok. I still think it could use a refactor, maybe introduce `unsafe_verify` instead of making the main `verify` function so easy to hack